### ISSUE TITLE
Introduce elliptic curve element type trait

### ISSUE THREAD

**Issue by nemothenoone** [NilFoundation/crypto3-algebra#2]
Concept introduction along with type traits formalization is required.

**Comment by nkaskov** [comment]
Closed by 4eadbad2a746041eeb7a0276bf8fe4acb2a79331

**Comment by nkaskov** [comment]
Depends on #1

**Comment by nkaskov** [comment]
Duplicate of #32
